aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -457,11 +457,10 @@ Tensor& addmm_out_sparse_compressed_cpu(
       mat1.size(1) == mat2.size(0), "mat1 and mat2 shapes cannot be multiplied (",
       mat1.size(0), "x", mat1.size(1), " and ", mat2.sizes()[0], "x", mat2.sizes()[1], ")");
 
-  if (mat1.layout() == kSparseCsc || mat2.layout() == kSparseCsc) {
+  if (mat1.layout() == kSparseCsc && mat2.layout() == kSparseCsc) {
```

Comment:
Does this change restrict the existing support matrix of `addmm`?

Reply:
No, this condition lead to some valid combinations getting blocked one layer deeper in  `at::native:sparse::impl::mkl::addmm_out_sparse_csr` where more layout handling is done.  

This check is poorly placed and it is too greedy. It forces unnecessary conversions and can cause some strange "layout combination not supported errors".  Suppose you pass `mat1(strided)` `mat2(csc)`, and `self/result (strided)`. The original check would catch causing a `strided->csr` conversion on `mat1`, and an unnecessary and expensive `csc->csr` conversion on `mat2`. When this eventually enters the `impl::mkl` layer you get an error that `strided + csr @ csr` is not supported. Instead now the `strided + strided @ CSC` combination is properly handled by evaluating the transposed operation which is `strided + CSR @ strided` which we have support for. 

In fact, coupled with the other changes in the `mkl` interface in this pr relaxing the condition here expands support for matrix addmm.

Reply:
Makes sense.